controllers/components/kserve/kserve_controller_actions.go

```diff
@@ -138,91 +138,160 @@ func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
 	}
 
-	for _, ftName := range ftNames {
-		obj := &featuresv1.FeatureTracker{}
-		err := rr.Client.Get(ctx, client.ObjectKey{Name: ftName}, obj)
-		switch {
-		case k8serr.IsNotFound(err):
+	for _, n := range ftNames {
+		ft := featuresv1.FeatureTracker{}
+		err := rr.Client.Get(ctx, client.ObjectKey{Name: n}, &ft)
+		if k8serr.IsNotFound(err) {
 			continue
-		case err != nil:
-			return fmt.Errorf("error while retrieving FeatureTracker %s: %w", ftName, err)
+		} else if err != nil {
+			return fmt.Errorf("failed to lookup FeatureTracker %s: %w", ft.GetName(), err)
 		}
 
-		if err := resources.RemoveOwnerReferences(ctx, rr.Client, obj, isLegacyOwnerRef); err != nil {
-			return err
+		err = rr.Client.Delete(ctx, &ft, client.PropagationPolicy(metav1.DeletePropagationForeground))
+		if k8serr.IsNotFound(err) {
+			continue
+		} else if err != nil {
+			return fmt.Errorf("failed to delete FeatureTracker %s: %w", ft.GetName(), err)
 		}
 	}
 
 	return nil
 }
 
-func configureServerless(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+func addServingCertResourceIfManaged(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
 	k, ok := rr.Instance.(*componentApi.Kserve)
 	if !ok {
 		return fmt.Errorf("resource instance %v is not a componentApi.Kserve)", rr.Instance)
 	}
 
-	logger := logf.FromContext(ctx)
-	cli := rr.Client
-
-	switch k.Spec.Serving.ManagementState {
-	case operatorv1.Unmanaged: // Bring your own CR
-		logger.Info("Serverless CR is not configured by the operator, we won't do anything")
-
-	case operatorv1.Removed: // we remove serving CR
-		logger.Info("existing Serverless CR (owned by operator) will be removed")
-		if err := removeServerlessFeatures(ctx, rr.Client, k, &rr.DSCI.Spec); err != nil {
-			return err
-		}
-
-	case operatorv1.Managed: // standard workflow to create CR
+	if k.Spec.Serving.ManagementState == operatorv1.Managed {
 		if rr.DSCI.Spec.ServiceMesh == nil {
 			return errors.New("ServiceMesh needs to be configured and 'Managed' in DSCI CR, " +
 				"it is required by KServe serving")
 		}
 
-		switch rr.DSCI.Spec.ServiceMesh.ManagementState {
-		case operatorv1.Unmanaged, operatorv1.Removed:
-			return fmt.Errorf("ServiceMesh is currently set to '%s'. It needs to be set to 'Managed' in DSCI CR, "+
-				"as it is required by the KServe serving field", rr.DSCI.Spec.ServiceMesh.ManagementState)
+		err := createServingCertResource(ctx, rr.Client, &rr.DSCI.Spec, k)
+		if err != nil {
+			return fmt.Errorf("unable to create serverless serving certificate secret: %w", err)
 		}
+	}
+	return nil
+}
 
-		serverlessFeatures := feature.ComponentFeaturesHandler(rr.Instance, componentName, rr.DSCI.Spec.ApplicationsNamespace, configureServerlessFeatures(&rr.DSCI.Spec, k))
+func addTemplateFiles(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	// Always add all the templates...will get removed before deploy if necessary
+	templates := []odhtypes.TemplateInfo{
```

Comment:
@grdryn you may find this pr useful https://github.com/opendatahub-io/opendatahub-operator/pull/1757 (but no need to wait for it)

Reply:
That does look interesting! I'll try to find time to review, but like you said won't necessarily block this PR on it.